### PR TITLE
Adding updates to the CMake: Quick Start to the existing docs 

### PR DESCRIPTION
**Pull request by sinemakinci1**
Updating the docs to be up-to-date with the current CMake: Quick Start experience and breaking this out into its own page

### CODE REVIEW

File: docs/cpp/cmake-linux.md [hunk 1]
```diff
@@ -78,57 +78,47 @@ sudo apt-get install build-essential gdb
...
 1. Select the compiler you want to use. For example, depending on the compilers you have installed, you might see something like:
 
     ![Select the kit](images/cpp/cmake-selectkit.png)
 
-## Configure Hello World
-
-There are two things you must do to configure your CMake project: select a kit (which you just did) and select a variant.
-
-The kit you selected previously is shown in the **Project Status** view in the CMake Tools sidebar. For example:
+The kit you selected previously is shown in the **Project Status** section in the CMake Tools view. For example:
```
Comment (ntrogh) [inline docs/cpp/cmake-linux.md:115]:
```suggestion
    The kit you selected previously is now shown in the **Project Status** section in the CMake Tools view.
```

File: docs/cpp/cmake-linux.md [hunk 2]
```diff
@@ -78,57 +78,47 @@ sudo apt-get install build-essential gdb
...
 
 ![Selected kit in Sidebar](images/cpp/cmake-kit.png)
```
Comment (ntrogh) [inline docs/cpp/cmake-linux.md:117]:
```suggestion
    ![Selected kit in Sidebar](images/cpp/cmake-kit.png)
```

File: docs/cpp/cmake-linux.md [hunk 3]
```diff
@@ -78,57 +78,47 @@ sudo apt-get install build-essential gdb
...
 
-To change the kit, you can click on the kit in the **Project Status** view in the CMake Tools sidebar, or run the **CMake: Select a Kit** command again from the Command Palette. If you don't see the compiler you're looking for, you can edit the `cmake-tools-kits.json` file in your project. To edit the file, open the Command Palette (`kb(workbench.action.showCommands)`) and run the **CMake: Edit User-Local CMake Kits** command.
+To change the kit, you can select the kit in the **Project Status** section in the CMake Tools view, or run the **CMake: Select a Kit** command again from the Command Palette. If you don't see the compiler you're looking for, you can edit the `cmake-tools-kits.json` file in your project. To edit the file, open the Command Palette (`kb(workbench.action.showCommands)`) and run the **CMake: Edit User-Local CMake Kits** command.
 
-### Select a variant
+##### Select a variant
```
Comment (ntrogh) [inline docs/cpp/cmake-linux.md:121]: Is this section a subsection of "Configure using CMake Kits"? It's weird to have a single subsection.

File: docs/cpp/cmake-linux.md [hunk 4]
```diff
@@ -78,57 +78,47 @@ sudo apt-get install build-essential gdb
...
+![CMake Side Panel with presets](images/cpp/cmake-side-panel-presets.png)
 
-**Note:** If you had wanted to create a basic source and header file, you would have selected **Library** instead. But for this tutorial, **Executable** will do. If you are prompted to configure IntelliSense for the folder, select **Allow**.
+You can also set any of your presets by running the **CMake: Select Configure Preset** or **CMake: Select Build Preset** commands in the Command Palette (`kb(workbench.action.showCommands)`).
 
-This creates a hello world CMake project containing `main.cpp`, `CMakeLists.txt` (which tells the CMake tools how to build your project), and a folder named `build` for your build files:
+![CMake Select presets](images/cpp/select-cmake-preset.png)
 
-![Project contents](images/cpp/cmake-project-contents.png)
+#### Configure Using CMake Kits
```
Comment (ntrogh) [inline docs/cpp/cmake-linux.md:104]:
```suggestion
#### Configure using CMake Kits
```

File: docs/cpp/cmake-linux.md [hunk 5]
```diff
@@ -78,57 +78,47 @@ sudo apt-get install build-essential gdb
...
-mkdir cmakeQuickStart
-cd cmakeQuickStart
-code .
-```
-
-The `code .` command opens VS Code in the current working folder, which becomes your "workspace".
+### Configure Hello World
 
-### Create a CMake hello world project
+Before you can use the CMake Tools extension to build a project, you need to configure it to know about the compilers on your system. There are two ways to configure CMake in VS Code: CMake Presets and CMake Kits/Variants.
```
Comment (ntrogh) [inline docs/cpp/cmake-linux.md:87]:
```suggestion
Before you can use the CMake Tools extension to build a project, you need to configure it to know about the compilers on your system. There are two ways to configure CMake in VS Code: 

- Use CMake Presets (recommended)
- Use CMake Kits/Variants
```

File: docs/cpp/cmake-linux.md [hunk 6]
```diff
@@ -78,57 +78,47 @@ sudo apt-get install build-essential gdb
 
 ## Create a CMake project
 
-If you have an existing CMake project that already has a `CMakeLists.txt` file in the root directory, you can skip to [Select a kit](#select-a-kit) to configure your existing project.
+If you do not have an existing CMake project, follow the steps in [Create a CMake project](cmake-quickstart.md).
 
-Otherwise, create a folder for a new project. From the Terminal window, create an empty folder called `cmakeQuickStart`, navigate into it, and open VS Code in that folder by entering the following commands:
+If you already have an existing CMake project that has a `CMakeLists.txt` file in the root directory, skip to [Configure Hello World](#configure-hello-world) to configure your project.
```
Comment (ntrogh) [inline docs/cpp/cmake-linux.md:83]:
```suggestion
If you already have an existing CMake project that has a `CMakeLists.txt` file in the root directory, continue to [Configure Hello World](#configure-hello-world) to configure your project.
```

File: docs/cpp/cmake-quickstart.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+```
+The ```code .``` command opens VS Code in the current working folder, which becomes your "workspace".
+
+## Create a CMakeLists.txt file
+
+The CMake Tools extension can create the files for a basic CMake project for you. 
+
+1. Open the Command Palette (`kb(workbench.action.showCommands)`) and run the **CMake: Quick Start** command:
+
+  ![Create CMake quickstart](images/cpp/cmake-quickstart-command-palette.png)
```
Comment (ntrogh) [inline docs/cpp/cmake-quickstart.md:31]:
```suggestion
    ![Create CMake quickstart](images/cpp/cmake-quickstart-command-palette.png)
```

File: docs/cpp/cmake-quickstart.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+1. Enter a project name, and select and **C++** as the project language.
+
+  This information will be written to `CMakeLists.txt` and a few initial source files.
```
Comment (ntrogh) [inline docs/cpp/cmake-quickstart.md:35]:
```suggestion
    This information will be written to `CMakeLists.txt` and a few initial source files.
```

File: docs/cpp/cmake-quickstart.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+  > **Note:** If you had other source code files in this folder that you wanted to add as targets to the `CmakeLists.txt`, an option to add these would now be given. But for this tutorial, we will stick with just the hello world file.
```
Comment (ntrogh) [inline docs/cpp/cmake-quickstart.md:37]:
```suggestion
    > **Note:** If you had other source code files in this folder that you wanted to add as targets to the `CmakeLists.txt`, an option to add these would now be given. But for this tutorial, we will stick with just the hello world file.
```

File: docs/cpp/cmake-quickstart.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+1. Select **CTest** as an additional option to add support for testing. You can also select **CPack** for CPack support.
+
+  ![Additional Options](images/cpp/cmake-quickstart-options.png)
```
Comment (ntrogh) [inline docs/cpp/cmake-quickstart.md:41]:
```suggestion
    ![Additional Options](images/cpp/cmake-quickstart-options.png)
```

File: docs/cpp/cmake-quickstart.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+1. Next, select **Executable** as the project type to create a basic source file (`main.cpp`) that includes a basic `main()` function.
+
+  ![Choose project type](images/cpp/cmake-choose-type.png)
```
Comment (ntrogh) [inline docs/cpp/cmake-quickstart.md:45]:
```suggestion
    ![Choose project type](images/cpp/cmake-choose-type.png)
```

File: docs/cpp/cmake-quickstart.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+  > **Note:** If you had wanted to create a basic source and header file, you would have selected **Library** instead. But for this tutorial, **Executable** will do. If you are prompted to configure IntelliSense for the folder, select **Allow**.
```
Comment (ntrogh) [inline docs/cpp/cmake-quickstart.md:47]:
```suggestion
    > **Note:** If you had wanted to create a basic source and header file, you would have selected **Library** instead. But for this tutorial, **Executable** will do. If you are prompted to configure IntelliSense for the folder, select **Allow**.
```

File: docs/cpp/cmake-quickstart.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+
+![Project contents](images/cpp/cmake-quickstart-cmakelists.png)
+
+## Create a CMakePresets.json file
+
+Next, continue with the CMake Quick Start to create a `CMakePresets.json` file.
+
+1. Select **Add a New Preset** and **Create from Compilers**.
+
+  The extension automatically scans for kits on your computer and creates a list of compilers found on your system.
```
Comment (ntrogh) [inline docs/cpp/cmake-quickstart.md:59]:
```suggestion
    The extension automatically scans for kits on your computer and creates a list of compilers found on your system.
```

File: docs/cpp/cmake-quickstart.md [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+
+1. Select the compiler you want to use.
+
+  For example, depending on the compilers you have installed, you might see something like this:
```
Comment (ntrogh) [inline docs/cpp/cmake-quickstart.md:63]:
```suggestion
    For example, depending on the compilers you have installed, you might see something like this:
```

File: docs/cpp/cmake-quickstart.md [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+
+  ![Add a new preset](images/cpp/cmake-quickstart-selectkit.png)
```
Comment (ntrogh) [inline docs/cpp/cmake-quickstart.md:65]:
```suggestion
    ![Add a new preset](images/cpp/cmake-quickstart-selectkit.png)
```

File: docs/cpp/cmake-quickstart.md [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+
+1. Enter a name for this new preset. This will be written to `CMakePresets.json`.
```
Comment (ntrogh) [inline docs/cpp/cmake-quickstart.md:67]:
```suggestion
1. Enter a name for this new preset.

    The name for the preset will be written to `CMakePresets.json`.
```

File: docs/cpp/cmake-quickstart.md (hunk 13 above)
Comment (ntrogh) [inline docs/cpp/cmake-quickstart.md:59]:
```suggestion
    The extension automatically scans for kits on your computer and creates a list of compilers found on your system.
```

File: docs/cpp/cmake-quickstart.md [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+ContentId: 55b5d15c-a020-4808-941f-e0255751a5f7
+PageTitle: Create a CMake project with the CMake Quick Start
+DateApproved: 5/29/2024
+MetaDescription: Create a Hello World project by using the CMake Quick Start in the CMake Tools Visual Studio Code extension
+---
+# Create a CMake hello world project with CMake Quick Start
+
+In this article, you'll learn how to create a CMake hello world project from scratch using the CMake Tools extension in VS Code.
+
+If you have an existing CMake project that already has a `CMakeLists.txt` file in the root directory but no CMake presets, you can skip to [Create a CmakePresets.json file](#Create-a-CMakePresets.json-file) to configure your project with CMake presets.
```
Comment (ntrogh) [inline docs/cpp/cmake-quickstart.md:14]:
```suggestion
If you have an existing CMake project that already has a `CMakeLists.txt` file in the root directory but no CMake presets, you can skip to [Create a `CMakePresets.json` file](#Create-a-CMakePresets.json-file) to configure your project with CMake presets.
```

**Review by ntrogh — COMMENTED** [review]
Some additional formatting fixes. Noticed that the indentation in the quick start was off.
We should be good after this pass.

File: docs/cpp/cmake-linux.md (hunk 3 above)
Comment (sinemakinci1) [reply docs/cpp/cmake-linux.md]: I added it to be one large section

**Comment by sinemakinci1** [comment]
@ntrogh ready for a final review!

**Review by ntrogh — APPROVED** [review]
Looks good to me! Ready to be merged.

File: docs/cpp/cmake-quickstart.md [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+If you have an existing CMake project that already has a `CMakeLists.txt` file in the root directory but no CMake presets, you can skip to [Create a `CMakePresets.json` file](#Create-a-CMakePresets.json-file) to configure your project with CMake presets.
+
+Otherwise, create a folder for a new project. From the Terminal window, create an empty folder called `HelloWorld`, navigate into it, and open VS Code in that folder by entering the following commands:
+
+```bash
+mkdir helloworld
+cd helloworld
+code .
+```
+The ```code .``` command opens VS Code in the current working folder, which becomes your "workspace".
```
Comment (roblourens) [inline docs/cpp/cmake-quickstart.md:23]: Maybe this should be a single-backtick codespan? Not sure how it's rendered on the website.

**Review by roblourens — DISMISSED** [review]

File: docs/cpp/cmake-quickstart.md (hunk 18 above)
Comment (ntrogh) [reply docs/cpp/cmake-quickstart.md:23]: Not sure why I suggested that.. Should indeed be single-backtick!

**Comment by sinemakinci1** [comment]
@roblourens @ntrogh Updated the backticks!

**Pull request merged**